src/VulkanSample.cpp

```diff
@@ -261,16 +262,30 @@ struct CommandLineParameters
     }
 } g_CommandLineParameters;
 
-void SetDebugUtilsObjectName(VkObjectType type, uint64_t handle, const char* name)
+void SetDebugUtilsObjectName(VkObjectType type, uint64_t handle, std::string name)
 {
-    if(vkSetDebugUtilsObjectNameEXT_Func == nullptr)
+    TEST(!name.empty());
+
+    // We must make sure the std::string we pass as name is still valid memory until the program ends.
```

Comment:
Why do you assume this is the case? Which part of the Vulkan specification says the string we pass to `vkSetDebugUtilsObjectNameEXT` must be alive for the whole lifetime of the object? I can't see any mention of it here:
https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkDebugUtilsObjectNameInfoEXT.html
I think this is not true, the string doesn't need to persist in memory, and the VK_EXT_debug_utils Vulkan extensions makes an internal copy of the string.

Reply:
Oh right, it does create a copy internally. I changes this.